packages/client/src/schedule-types.ts

```diff
@@ -0,0 +1,830 @@
+// type StartSomethingElseAction<Action extends SomethingElse> = ExpectsSomethingElse<Action>
+// type ScheduleActionOptions<Action extends ScheduleActionType> =
+//   StartWorkflowAction<Action> | StartSomethingElseAction<Action>
+
+/**
+ * Options for starting a Workflow
+ */
+export interface ScheduleOptions<Action extends ScheduleActionType> {
+  /**
+   * Schedule Id
+   *
+   * We recommend using a meaningful business identifier.
+   */
+  scheduleId: string;
+
+  /** When Actions should be taken */
+  spec: RequireAtLeastOne<ScheduleSpec, 'calendars' | 'intervals' | 'cronExpressions'>;
+
+  /**
+   * Which Action to take
+   */
+  action: ScheduleActionOptions<Action>;
+
+  /**
+   * Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+   * running. This can be changed after a Schedule has taken some Actions, and some changes might produce
+   * unintuitive results. In general, the later policy overrides the earlier policy.
+   *
+   * @default {@link ScheduleOverlapPolicy.SKIP}
+   */
+  overlap?: ScheduleOverlapPolicy;
+
+  /**
+   * The Temporal Server might be down or unavailable at the time when a Schedule should take an Action. When the Server
+   * comes back up, `catchupWindow` controls which missed Actions should be taken at that point. The default is one
+   * minute, which means that the Schedule attempts to take any Actions that wouldn't be more than one minute late. It
+   * takes those Actions according to the {@link ScheduleOverlapPolicy}. An outage that lasts longer than the Catchup
+   * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+   *
+   * @default 1 minute
+   * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+   */
+  catchupWindow?: number | string;
+
+  /**
+   * When an Action times out or reaches the end of its Retry Policy, {@link pause}.
+   *
+   * With {@link ScheduleOverlapPolicy.ALLOW_ALL}, this pause might not apply to the next Action, because the next Action
+   * might have already started previous to the failed one finishing. Pausing applies only to Actions that are scheduled
+   * to start after the failed one finishes.
+   *
+   * @default false
+   */
+  pauseOnFailure?: boolean;
+
+  /**
+   * Informative human-readable message with contextual notes, e.g. the reason
+   * a Schedule is paused. The system may overwrite this message on certain
+   * conditions, e.g. when pause-on-failure happens.
+   */
+  note?: string;
+
+  /**
+   * Start in paused state.
+   *
+   * @default false
+   */
+  paused?: boolean;
+
+  /**
+   * Limit the number of Actions to take.
+   *
+   * This number is decremented after each Action is taken, and Actions are not
+   * taken when the number is `0` (unless {@link ScheduleHandle.trigger} is called).
+   *
+   * @default unlimited
+   */
+  remainingActions?: number;
+
+  /**
+   * Trigger one Action immediately.
```

Comment:
might want to note that this is "Trigger one Action immediately on schedule creation" (same for backfill)